src/rascal/structure_managers/adaptor_increase_maxorder.hh

```diff
@@ -459,28 +430,84 @@ namespace rascal {
   void AdaptorMaxOrder<ManagerImplementation>::update_self_helper() {
     static_assert(traits::MaxOrder > 2,
                   "No neighbourlist present; extension not possible.");
+    using AddOrderLoop =
+        AddOrderLoop<2, traits::NeighbourListType, IsCompactCluster, true>;
+    static constexpr bool has_center_pair_order_two{traits::HasCenterPair and
+                                                    traits::MaxOrder - 1 == 2};
+    // size_t turns out to give 0 if false and 1 if true. When adding Order==3
+    // to the manager the input manager could have center pairs and
+    // cluster_start avoid iterating over those
+    static constexpr size_t cluster_start{
+        static_cast<size_t>(has_center_pair_order_two)};
 
     internal::for_each(this->cluster_indices_container,
                        internal::ResizePropertyToZero());
 
+    auto & atom_cluster_indices{std::get<0>(this->cluster_indices_container)};
+
     this->nb_neigh.clear();
     this->offsets.clear();
     this->neighbours_atom_tag.clear();
 
-    // #BUG8486@(markus) I now append the ghost atoms to the cluster index
-    // container
     for (auto atom : this->manager) {
-      //  Order 1, but variable Order is at 0, atoms, index 0
-      using AddOrderLoop = AddOrderLoop<atom.order(), traits::NeighbourListType,
-                                        IsCompactCluster,
-                                        atom.order() == (traits::MaxOrder - 1)>;
-
-      auto & atom_cluster_indices{std::get<0>(this->cluster_indices_container)};
-
-      auto indices = atom.get_cluster_indices();
+      // copy the underlying cluster indices of this->manager (not changed)
+      // do order == 1
+      auto indices{atom.get_cluster_indices()};
       atom_cluster_indices.push_back(indices);
-
-      AddOrderLoop::loop(atom, atom, 0, *this);
+      // do the rest
+      AddOrderLoop::loop(atom, *this);
+
+      this->add_entry_number_of_neighbours();
+      std::array<int, traits::MaxOrder> new_tag_list{};
+      // loop over the highest order available in this->manager
+      for (auto cluster :
+           atom.template get_clusters_of_order<traits::MaxOrder - 1>(
+               cluster_start)) {
+        auto && tag_list{cluster.get_atom_tag_list()};
+        // copy the tags from the previous order skiping the center atom tag
```

Comment:
skipPing.
I am not following the logic here - is the skip not on the first index? That would be the center atoms, right?